dissect/target/filesystem.py

```diff
@@ -1238,6 +1242,32 @@ def map_file_entry(self, vfspath: str, entry: FilesystemEntry) -> None:
             entry_name = fsutil.basename(vfspath, alt_separator=self.alt_separator)
             directory.add(entry_name, entry)
 
+    def map_dir_from_tar(self, vfspath: str, tar_file: Union[str, pathlib.Path], custom_path: bool = False) -> None:
```

Comment:
```suggestion
    def map_dir_from_tar(self, vfspath: str, tar_file: str | pathlib.Path, custom_path: bool = False) -> None:
```
In spirit of my comment from the Docker PR.